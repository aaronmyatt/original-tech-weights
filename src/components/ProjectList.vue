<template>
<ul class="project-list">
  <li class="project card" v-for="project, index in projectList" :key="index">
    <Project v-bind="project" />
  </li>
</ul>
</template>

<script>
import Project from './Project';

export default {
  name: 'ProjectList',
  props: ['projectList'],
  components: {
    Project,
  },
};
</script>

<style scoped>
</style>
