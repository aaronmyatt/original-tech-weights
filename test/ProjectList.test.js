import { shallow } from 'vue-test-utils'
import ProjectList from '../src/components/ProjectList'

describe('ProjectList', () => {
  let cmp

  beforeEach(() => {
    cmp = shallow(ProjectList, {
        propsData: {
            projectList: ['Test project'],
        },
    })
  })

  it('has the expected html structure', () => {
    expect(cmp.element).toMatchSnapshot()
  })

  describe('Props', () => {

      it('renders a list of projects', () => {
          expect(cmp.findAll('.project-list .project')).toHaveLength(cmp.vm.projectList.length)
      })

      it('renders a all projects in the list', () => {
          cmp.vm.projectList.push('Another Test Project')
          cmp.update()
          expect(cmp.findAll('.project-list .project')).toHaveLength(cmp.vm.projectList.length)
      })

  })

})
