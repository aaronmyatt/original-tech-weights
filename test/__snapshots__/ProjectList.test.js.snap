// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectList has the expected html structure 1`] = `
<ul
  class="project-list"
>
  <li
    class="project card"
  >
    <!---->
  </li>
</ul>
`;
